### PR TITLE
Untangle the server / handler registration a bit.

### PR DESCRIPTION
**Pull request by Martin2112**
Remove RegisterHandler and have the servers do the same thing with one
less layer of indirection. Server no longer refers to HttpHandler purely
for this.

### CODE REVIEW

File: cpp/server/server.cc [hunk 1]
```diff
@@ -191,6 +183,14 @@ ContinuousFetcher* Server::continuous_fetcher() {
   return fetcher_.get();
 }
 
+Proxy* Server::proxy() {
```
Comment (pphaneuf) [inline cpp/server/server.cc]: Needs two lines between top level declarations/definitions.

File: cpp/server/server.cc [hunk 2]
```diff
@@ -191,6 +183,14 @@ ContinuousFetcher* Server::continuous_fetcher() {
...
+  return proxy_.get();
+}
+
```
Comment (pphaneuf) [inline cpp/server/server.cc]: Here too.

**Comment by pphaneuf** [comment]
You might be able to get rid of the `#include "server/handler.h"` in `server.cc`, no? Otherwise done with the review, just those nits. :smiley:

**Comment by Martin2112** [comment]
All done.

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**